bpf/process/types/basic.h

```diff
@@ -1636,7 +1636,16 @@ FUNC_INLINE int match_binaries(__u32 selidx)
 			if (!postfix_key)
 				return 0;
 			postfix_key->prefixlen = postfix_len * 8; // prefixlen is in bits
-			file_copy_reverse(postfix_key->data, postfix_len, (__u8 *)current->bin.path, current->bin.path_length - postfix_len);
+			if (!current->bin.reversed) {
+				file_copy_reverse((__u8 *)current->bin.end_r, postfix_len, (__u8 *)current->bin.path, current->bin.path_length - postfix_len);
```

Comment:
Wasn't the goal of keeping `end` to reverse `bin.end` and not `bin.path`? I think adding a test with a long binary like in 67348f6f826cc68c40625dbdeb8cc90dec5bc455 would have caught that.

Reply:
Oh, thanks, my fault).